src/node_jsvmapi.cc

```diff
@@ -496,6 +492,48 @@ namespace v8impl {
 
 }  // end of namespace v8impl
 
+// Intercepts the Node-V8 module registration callback. Converts parameters to NAPI equivalents
+// and then calls the registration callback specified by the NAPI module.
+void napi_module_register_cb(v8::Local<v8::Object> exports,
+                             v8::Local<v8::Value> module,
+                             v8::Local<v8::Context> context,
+                             void* priv) {
+  napi_module* mod = static_cast<napi_module*>(priv);
+  mod->nm_register_func(
+    v8impl::JsEnvFromV8Isolate(context->GetIsolate()),
+    v8impl::JsValueFromV8LocalValue(exports),
+    v8impl::JsValueFromV8LocalValue(module),
+    mod->nm_priv);
+}
+
+namespace node {
+  // Indicates whether NAPI was enabled/disabled via the node command-line.
+  extern bool no_napi_modules;
+}
+
+// Registers a NAPI module.
+void napi_module_register(napi_module* mod) {
+  // NAPI modules always work with the current node version.
+  int moduleVersion = NODE_MODULE_VERSION;
+  if (node::no_napi_modules) {
+    // NAPI is dsabled, so set the module version to -1 to cause the module to be unloaded.
+    moduleVersion = -1;
+  }
+
+  static node::node_module nm = {
+    moduleVersion,
+    mod->nm_flags,
+    nullptr,
```

Comment:
No, I don't think so. The 3 `nullptr` fields are:
 * `nm_dso_handle` - used internally by node to store the uv module handle; it is never provided via the module registration func.
 * `nm_register_func` - older signature of the register func (`addon_register_func`). We don't need to use it because we use the newer signature (`addon_context_register_func`) that takes the V8 context (and isolate) as a parameter.
 * `nm_link` - used internally by node to create a linked-list of modules; it is never provided by the module registration func.